wagtail/models/pages.py

```diff
@@ -967,14 +971,37 @@ def save_revision(
             # We need to ensure comments have an id in the revision, so positions can be identified correctly
             comment.save()
 
-        revision = Revision.objects.create(
-            content_object=self,
-            base_content_type=self.get_base_content_type(),
-            user=user,
-            approved_go_live_at=approved_go_live_at,
-            content=self.serializable_data(),
-            object_str=str(self),
-        )
+        if overwrite_revision:
+            # the revision being overwritten must be the latest revision for the current page, and must match
+            # the current user (if any)
+            latest_revision = self.get_latest_revision()
+            if overwrite_revision != latest_revision:
+                raise PermissionDenied(
+                    "Cannot overwrite a revision that is not the latest for this page"
+                )
+
+            if (user is None and overwrite_revision.user_id is not None) or (
+                user is not None and overwrite_revision.user_id != user.pk
+            ):
```

Comment:
I think this is equivalent, though it might be less readable. If so I'm happy with leaving it as-is

```suggestion
            if overwrite_revision.user_id != (user and user.pk):
```